api_app/views.py

```diff
@@ -574,3 +587,37 @@ class TagViewSet(viewsets.ModelViewSet):
     queryset = Tag.objects.all()
     serializer_class = TagSerializer
     pagination_class = None
+
+
+class UserCustomConfigViewSet(viewsets.ModelViewSet):
+    queryset = CustomConfig.objects.filter(organization__isnull=True)
+    serializer_class = CustomConfigSerializer
+
+    def get_queryset(self):
+        return self.queryset.filter(owner=self.request.user)
+
+    def perform_create(self, serializer):
+        serializer.save(owner=self.request.user, organization=None)
+
+    def perform_update(self, serializer):
+        serializer.save(owner=self.request.user, organization=None)
+
+    # check for duplicate CustomConfig
+    def create(self, request, *args, **kwargs):
+        try:
+            CustomConfig.objects.get(
+                owner=request.user,
+                name=request.data["name"],
+                organization=None,
+                attribute=request.data["attribute"],
+                type=request.data["type"],
+            )
+        except CustomConfig.DoesNotExist:
+            return super().create(request, *args, **kwargs)
+        else:
+            raise ValidationError(
+                {
+                    "detail": "A custom config for this entity "
+                    "already exists with same scope."
+                }
+            )
```

Comment:
ok but this is a validation so this should be implemented in the related serializer